File: wasm/wasi/fibonacci.zig
const std = @import("std");

fn fibonacci(n: u32) u32 {
    if (n == 0) return 0;
    if (n == 1) return 1;
    return fibonacci(n - 1) + fibonacci(n - 2);
}

pub fn main() !void {
    const stdout = std.io.getStdOut().writer();
    var x: u32 = 5;

    try stdout.print("fibonacci({}) = {}\n", .{ x, fibonacci(x) });
}

File: wasm/wasi/fibonacci.js
const { readFileSync } = require("node:fs");
const { WASI, init } = require("@wasmer/wasi");

(async () => {
  await init();

  const wasi = new WASI({
    env: {},
    args: [
      "fibonacci",
      "10",
    ],
  });

  const buffer = readFileSync("./fibonacci.wasm");

  const module = await WebAssembly.compile(new Uint8Array(buffer));

  await wasi.instantiate(module, {});

  const exitCode = wasi.start();
  const stdout = wasi.getStdoutString();

  console.log(`${stdout}(exit code: ${exitCode})`);
})();
